data-final-fixes.lua

```diff
@@ -107,17 +107,22 @@ shared.drone_collision_mask ={layers = all_used_tile_collision_masks}
 shared.drone_collision_mask["colliding-with-tiles-only"] = true
 shared.drone_collision_mask["consider-tile-transitions"] = true
 --temp
-
-for k, prototype in pairs(collision_mask_util.collect_prototypes_with_layer("player")) do
-  if prototype.type ~= "gate" and prototype.type ~= "tile" then
-    local mask = collision_mask_util.get_mask(prototype)
-    if mask.layers["item"] then
-      mask.layers["roadtd"] = true
-    end
-    prototype.collision_mask = mask
+collision_mask_util.collect_prototypes_with_layer = function(layer)
+  local prototype_list = {}
+  for type, default_mask in pairs(default_masks) do
+      if data.raw[type] then
+          for name, entity in pairs(data.raw[type]) do
+              local entity_mask = entity.collision_mask or default_mask
+              if entity_mask.layers and entity_mask.layers[layer] then
+                  table.insert(prototype_list, entity)
+              end
+          end
+      end
```

Comment:
Fix copied from [this](https://mods.factorio.com/mod/Transport_Drones_Meglinge_Fork/discussion/6778acf3f222dda95da53f88) thread